### PR TITLE
ES-857 

### PR DESCRIPTION
**Pull request by panharith-0118**
parent ticket https://mosip.atlassian.net/jira/software/c/projects/ES/boards/92?selectedIssue=ES-857
sub tickets
https://mosip.atlassian.net/jira/software/c/projects/ES/boards/92?selectedIssue=ES-886
https://mosip.atlassian.net/jira/software/c/projects/ES/boards/92?selectedIssue=ES-890
https://mosip.atlassian.net/jira/software/c/projects/ES/boards/92?selectedIssue=ES-887
https://mosip.atlassian.net/jira/software/c/projects/ES/boards/92?selectedIssue=ES-884
https://mosip.atlassian.net/jira/software/c/projects/ES/boards/92?selectedIssue=ES-885
https://mosip.atlassian.net/jira/software/c/projects/ES/boards/92?selectedIssue=ES-888
https://mosip.atlassian.net/jira/software/c/projects/ES/boards/92?selectedIssue=ES-892

### CODE REVIEW

**Review by rathanak-0080 — APPROVED** [review]

File: signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java [hunk 1]
```diff
@@ -142,7 +142,7 @@ public GenerateChallengeResponse generateChallenge(GenerateChallengeRequest gene
         if(cacheUtilService.isIdentifierBlocked(identifier))
             throw new SignUpException(ErrorConstants.IDENTIFIER_BLOCKED);
 
-        if(!generateChallengeRequest.isRegenerate()) {
+        if(!generateChallengeRequest.isRegeneratedChallenge()) {
```
Comment (bunsy-0900) [inline signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java:145]: Naming `isRegeneratedChallenge()` is good but `isRegeneratingNewChallange()` or `isRegenerateNewChallange()`  is also another good opt IMO.

**Review by bunsy-0900 — APPROVED** [review]

File: signup-service/src/main/java/io/mosip/signup/helper/NotificationHelper.java [hunk 2]
```diff
@@ -33,11 +34,14 @@ public class NotificationHelper {
...
+    @Value("#{${mosip.signup.supported-languages}}")
+    private List<String> supportedLanguages;
+
     public RestResponseWrapper<NotificationResponse> sendSMSNotification
             (String number, String locale, String templateKey, Map<String, String> params){
 
-        locale = locale != null ? locale : "khm";
-        String message = locale.equalsIgnoreCase("eng") ?
+        locale = locale != null ? locale : supportedLanguages.get(0);
+        String message = locale.equalsIgnoreCase(supportedLanguages.get(1)) ?
```
Comment (ase-101) [inline signup-service/src/main/java/io/mosip/signup/helper/NotificationHelper.java:44]: We should not assume that the secondary language notification template is not encoded always. Can we take this from properties?
something like this? mosip.signup.sms-notification-template.forgot-password.encoded.khm

File: signup-service/src/main/java/io/mosip/signup/services/GoogleRecaptchaValidatorService.java [hunk 3]
```diff
@@ -30,7 +30,7 @@ public class GoogleRecaptchaValidatorService implements CaptchaValidator {
     @Autowired
     private RestTemplate restTemplate;
 
-    @Timed(value = "validatecaptcha.api.timer", percentiles = {0.95, 0.99})
+    @Timed(value = "validatecaptcha.api.timer", percentiles = {0.9, 0.99})
     @Override
```
Comment (ase-101) [inline signup-service/src/main/java/io/mosip/signup/services/GoogleRecaptchaValidatorService.java:34]: Everywhere lets capture 90th percentile, we can remove 99th or 95th percentile

File: signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java [hunk 4]
```diff
@@ -251,21 +251,17 @@ public RegisterResponse register(RegisterRequest registerRequest, String transac
         transaction.setRegistrationStatus(RegistrationStatus.PENDING);
         cacheUtilService.setStatusCheckTransaction(transactionId, transaction);
 
-        notificationHelper.sendSMSNotificationAsync(registerRequest.getUserInfo().getPhone(), transaction.getLocale(),
-                        REGISTRATION_SMS_NOTIFICATION_TEMPLATE_KEY, null)
-                .thenAccept(notificationResponseRestResponseWrapper ->
-                    log.debug(notificationLogging, notificationResponseRestResponseWrapper)
-                );
+        notificationHelper.sendSMSNotification(registerRequest.getUserInfo().getPhone(), transaction.getLocale(),
```
Comment (ase-101) [inline signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java:254]: During registration, it should be ok to send the notification in a separate thread and not wait until the notification is sent out to complete and return the registration response.

File: signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java [hunk 5]
```diff
@@ -301,7 +301,10 @@ public RegistrationStatusResponse updatePassword(ResetPasswordRequest resetPassw
         restRequest.setRequesttime(IdentityProviderUtil.getUTCDateTime());
         restRequest.setRequest(identityRequest);
 
-        log.debug("Transaction {} : start reset password", transactionId);
+        return restRequest;
+    }
+
+    private void validateResetPasswordRequest(RestRequestWrapper restRequest, String transactionId ) {
```
Comment (ase-101) [inline signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java:307]: The method name is validateResetPasswordRequest but the method is only sending a patch request to the ID Repo server. We should rename the method.

File: signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java [hunk 6]
```diff
@@ -318,17 +321,23 @@ public RegistrationStatusResponse updatePassword(ResetPasswordRequest resetPassw
...
                 "phone", resetPasswordRequest.getIdentifier()), RegistrationStatus.PENDING);
         transaction.setRegistrationStatus(RegistrationStatus.PENDING);
         cacheUtilService.setStatusCheckTransaction(transactionId, transaction);
 
-        notificationHelper.sendSMSNotificationAsync(resetPasswordRequest.getIdentifier(), transaction.getLocale(),
-                        FORGOT_PASSWORD_SMS_NOTIFICATION_TEMPLATE_KEY, null)
-                .thenAccept(notificationResponseRestResponseWrapper ->
-                    log.debug(notificationLogging, notificationResponseRestResponseWrapper)
-                );
+        notificationHelper.sendSMSNotification(resetPasswordRequest.getIdentifier(), transaction.getLocale(),
```
Comment (ase-101) [inline signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java:339]: During reset-pwd, it should be ok to send the notification in a separate thread and not wait until the notification is sent out to complete and return the response.

File: signup-service/src/main/java/io/mosip/signup/dto/GenerateChallengeRequest.java [hunk 7]
```diff
@@ -13,7 +13,7 @@ public class GenerateChallengeRequest {
     private String captchaToken;
     @Language(required = false)
     private String locale;
-    private boolean regenerate;
+    private boolean regeneratedChallenge;
```
Comment (ase-101) [inline signup-service/src/main/java/io/mosip/signup/dto/GenerateChallengeRequest.java:16]: regenerateChallenge

File: signup-service/src/main/java/io/mosip/signup/helper/NotificationHelper.java [hunk 8]
```diff
@@ -33,13 +33,17 @@ public class NotificationHelper {
     @Value("${mosip.signup.send-notification.endpoint}")
     private String sendNotificationEndpoint;
 
+    @Value("{${mosip.signup.default-language}}")
```
Comment (ase-101) [inline signup-service/src/main/java/io/mosip/signup/helper/NotificationHelper.java:36]: Why are the extra flower braces added?

File: signup-service/src/main/resources/application-default.properties [hunk 9]
```diff
@@ -146,11 +147,13 @@ mosip.signup.ui.config.key-values={\
 ## ----------------------------- Notification templates ----------------------------------------------------------------
 # Default charset encoding ISO-8859-1 does not support khmer language characters, so templates in khm language are base64 encoded.
 mosip.signup.sms-notification-template.send-otp.khm=4Z6U4Z+S4Z6a4Z6+IHtjaGFsbGVuZ2V9IOGeiuGevuGemOGfkuGelOGeuOGeleGfkuGekeGfgOGehOGeleGfkuGekeGetuGej+Gfi+GeguGejuGek+GeuCBLaElEIOGemuGelOGen+Gfi+GeouGfkuGek+GegOGflA==
+mosip.signup.sms-notification-template.send-otp.khm.encode=true
```
Comment (ase-101) [inline signup-service/src/main/resources/application-default.properties:150]: Is it required to have a property for each template to check if the value is encoded?
can we create one property configured with the list of languages for which templates are encoded?

File: signup-service/src/main/resources/application-default.properties [hunk 10]
```diff
@@ -145,12 +146,12 @@ mosip.signup.ui.config.key-values={\
 
 ## ----------------------------- Notification templates ----------------------------------------------------------------
 # Default charset encoding ISO-8859-1 does not support khmer language characters, so templates in khm language are base64 encoded.
+mosip.signup.sms-notification-template.khm.encode=true
```
Comment (ase-101) [inline signup-service/src/main/resources/application-default.properties:149]: mosip.signup.sms-notification-template.encoded-langcodes=khm,hin

**Review by ase-101 — APPROVED** [review]

**Pull request merged**